### DISCUSSION TITLE
Just published a new GH package. Have few questions

### DISCUSSION THREAD

**Discussion by mittalyashu** [community/community#26624] [Packages]
<p>I have just published a docker image on GitHub packages. (the project is public)</p>
<p>It is showing the package is private, how can I make it public so that other users can download it.</p>
<p>When I tried to pull the docker image locally, it asked for authentication.</p>
<p>Is there a limited bandwidth for GH Packages?</p>
<p>Can I entirely use GH Packages instead of Docker Hub</p>

**Comment by jcansdale** [comment]
<p>Hi <a class="mention" href="/mittalyashu">@mittalyashu</a>,</p>
<aside class="quote no-group" data-username="mittalyashu" data-post="1" data-topic="132521">
<div class="title">
<div class="quote-controls"></div>
 mittalyashu:</div>
<blockquote>
<p>It is showing the package is private, how can I make it public so that other users can download it.</p>
</blockquote>
</aside>
<p>Could you make sure <code>Public</code> is checked for your organization, see here:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization" target="_blank" rel="noopener nofollow ugc">Enabling GitHub Container Registry for your organization - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You should then be able to change your container package’s visibility to public like this:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/configuring-access-control-and-visibility-for-container-images#configuring-visibility-of-container-images-for-an-organization" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/configuring-access-control-and-visibility-for-container-images#configuring-visibility-of-container-images-for-an-organization" target="_blank" rel="noopener nofollow ugc">Configuring access control and visibility for container images - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I hope that helps!</p>

**Comment by mittalyashu** [comment]
<p>After making the package public and trying to open in incognito mode, <a href="https://github.com/orgs/logchimp/packages/container/package/logchimp" rel="noopener nofollow ugc">https://github.com/orgs/logchimp/packages/container/package/logchimp</a></p>
<p>The link doesn’t work.</p>

**Comment by mittalyashu** [comment]
<p><a class="mention" href="/jcansdale">@jcansdale</a>. What about bandwidth limits and pull the docker image locally, it asked for authentication even after making the package as public.</p>

**Comment by shayneburgess** [comment]
<p>Thanks for the post <a class="mention" href="/mittalyashu">@mittalyashu</a>! We are investigating why that container is not loading in the UI - as far as I can tell it should be.</p>
<p>I was able to pull that container locally unauthenticated. Can you post the output when you try?</p>
<p>Shayne Burgess<br>
Eng Mgr - GHCR</p>

**Comment by mittalyashu** [comment]
<p>Awesome, thanks.</p>
<p><a class="mention" href="/shayneburgess">@shayneburgess</a> Thought, I find it strange it took few hours to make the link accessible publicly.</p>
<p>Also, even after linking the package with repository, it’s still not showing inside the repository.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/logchimp/logchimp" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/29014463/181098735-ec137adc-df56-49b0-91a2-c36d452b2e62.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e692f0ce5d0b91e3a5a6e71c838ba89ef1b88d7b_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e692f0ce5d0b91e3a5a6e71c838ba89ef1b88d7b_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e692f0ce5d0b91e3a5a6e71c838ba89ef1b88d7b_2_1380x690.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e692f0ce5d0b91e3a5a6e71c838ba89ef1b88d7b_2_10x10.png"></div>

<h3><a href="https://github.com/logchimp/logchimp" target="_blank" rel="noopener nofollow ugc">logchimp/logchimp</a></h3>

<p>Track user feedback to build better products with LogChimp - logchimp/logchimp</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by shayneburgess** [comment]
<p>You’re welcome! We believe we know why that package isn’t showing incognito and are working on a fix that shouldn’t take long.</p>
<p>I can’t think any reason why the link being accessible wouldn’t be immediate. Can you tell me more about what you saw?</p>
<p>The feature to show linked packages in the repository view is not available yet, unfortunately, but it is planned.</p>
<p>Shayne</p>

**Comment by mittalyashu** [comment]
<aside class="quote no-group" data-username="shayneburgess" data-post="7" data-topic="132521">
<div class="title">
<div class="quote-controls"></div>
 shayneburgess:</div>
<blockquote>
<p>I can’t think any reason why the link being accessible wouldn’t be immediate. Can you tell me more about what you saw?</p>
</blockquote>
</aside>
<p>It showed 404 page.</p>
<aside class="quote no-group" data-username="shayneburgess" data-post="7" data-topic="132521">
<div class="title">
<div class="quote-controls"></div>
 shayneburgess:</div>
<blockquote>
<p>The feature to show linked packages in the repository view is not available yet, unfortunately, but it is planned.</p>
</blockquote>
</aside>
<p>On that note, is it possible (or please do consider) linking package to multiple repository.</p>
<p>My use-case is I have created a separate repository for build docker images and main app is in separate repository.</p>
<ul>
<li><a href="https://github.com/logchimp/logchimp" rel="noopener nofollow ugc">https://github.com/logchimp/logchimp</a></li>
<li><a href="https://github.com/logchimp/docker" rel="noopener nofollow ugc">https://github.com/logchimp/docker</a></li>
</ul>

**Comment by clarkbw** [comment]
<p>The source label, <code>org.opencontainers.image.source</code> is intended for the repository source of the Dockerfile. In your case it might make sense to use the <code>org.opencontainers.image.url</code> label for your main application. We don’t currently linkify that label on the container page but we could look into doing that.</p>

**Comment by clarkbw** [comment]
<p>This has been fixed! Try it out and let us know <a href="https://github.com/orgs/github/packages/container/package/super-linter">https://github.com/orgs/github/packages/container/package/super-linter</a></p>

**Comment by mittalyashu** [comment]
<aside class="quote group-github-employees" data-username="clarkbw" data-post="9" data-topic="132521">
<div class="title">
<div class="quote-controls"></div>
 clarkbw:</div>
<blockquote>
<p>The source label, <code>org.opencontainers.image.source</code> is intended for the repository source of the Dockerfile. In your case it might make sense to use the <code>org.opencontainers.image.url</code> label for your main application.</p>
</blockquote>
</aside>
<p>Is there any list for label?</p>

**Comment by clarkbw** [comment]
<p>Those labels aren’t designed to be lists. It would break compatibility with other registries.</p>

**Comment by airtower-luna** [comment]
<p>I think the question might’ve been about a list of labels the GHCR recognizes and uses, which I’d like to see as well. The documentation currently only describes <code>org.opencontainers.image.source</code> in <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/connecting-a-repository-to-a-container-image">Connecting a repository to a container image</a>. I noticed <code>org.opencontainers.image.description</code> is shown on the Github package page, but that isn’t documented as far as I can see.</p>

**Comment by clarkbw** [comment, marked as answer]
<p>Created an issue for docs here: <a href="https://github.com/github/docs/issues/2617" class="inline-onebox">Document all the GHCR labels which can be used · Issue #2617 · github/docs · GitHub</a></p>
